pkg/execution/state/redis_state/queue_processor.go

```diff
@@ -673,6 +664,18 @@ func (q *queue) processPartition(ctx context.Context, p *QueuePartition, shard *
 		telemetry.IncrPartitionGoneCounter(ctx, telemetry.CounterOpt{PkgName: pkgName})
 		return nil
 	}
+	if errors.Is(err, ErrAccountConcurrencyLimit) {
+		q.lifecycles.OnAccountConcurrencyLimitReached(context.WithoutCancel(ctx), p.AccountID)
+		// TODO(cdzombak): telemetry?
+		// TODO(cdzombak): requeue?
```

Comment:
cc @tonyhb: do we need/want to requeue here upon hitting account or custom key concurrency limits, like we do on hitting partition concurrency limit?

I think probably yes, to avoid churning on these partitions unnecessarily. right?

Reply:
I believe so. We should requeue in all the cases we split in `PartitionLease` (catch-all`case -1, -2, -3:` -> separate errors)